> Can you confirm this issues still exists on the latest ComfyUI? I am on 0.34.0 and not seeing the issue. If you are still having this issue, please provide a workflow so I can test further.

@Mossbraker confirmed, this is still an issue so the PR is still needed. As soon as I deleted my fork (with the PR) and went back to this https://github.com/xxmjskxx/ComfyUI_SaveImageWithMetaDataUniversal repo version, I received the following:
```
[INFO] Requested to load Krea2TEModel_
[INFO] Model Krea2TEModel_ prepared for dynamic VRAM loading. 8463MB Staged. 0 patches attached. Force pre-loaded 249 weights: 627 KB.
[INFO] Model Krea2TEModel_ prepared for dynamic VRAM loading. 8463MB Staged. 0 patches attached. Force pre-loaded 249 weights: 627 KB.
[INFO] Requested to load Krea2
[INFO] Model Krea2 prepared for dynamic VRAM loading. 12864MB Staged. 270 patches attached. Force pre-loaded 160 weights: 2824 KB.
100%|███████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:17<00:00,  1.80s/it]
[INFO] Requested to load WanVAE
[INFO] Model WanVAE prepared for dynamic VRAM loading. 242MB Staged. 0 patches attached. Force pre-loaded 56 weights: 33 KB.
[INFO] Requested to load Krea2
[INFO] Model Krea2 prepared for dynamic VRAM loading. 12864MB Staged. 270 patches attached. Force pre-loaded 160 weights: 2824 KB.
100%|█████████████████████████████████████████████████████████████████████████████████████████████████| 4/4 [00:07<00:00,  1.82s/it]
[INFO] Requested to load WanVAE
[INFO] Model WanVAE prepared for dynamic VRAM loading. 242MB Staged. 0 patches attached. Force pre-loaded 56 weights: 33 KB.

0: 640x512 (no detections), 10.9ms
Speed: 2.2ms preprocess, 10.9ms inference, 1.3ms postprocess per image at shape (1, 3, 640, 512)
[INFO] [Metadata Loader] Refreshing definitions (defaults + ext, then conditional user JSON)...
[ERROR] !!! Exception during processing !!! unhashable type: 'dict'
[ERROR] Traceback (most recent call last):
  File "D:\comfy\execution.py", line 545, in execute
    output_data, output_ui, has_subgraph, has_pending_tasks = await get_output_data(prompt_id, unique_id, obj, input_data_all, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb, v3_data=v3_data)
                                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\comfy\execution.py", line 344, in get_output_data
    return_values = await _async_map_node_over_list(prompt_id, unique_id, obj, input_data_all, obj.FUNCTION, allow_interrupt=True, execution_block_cb=execution_block_cb, pre_execute_cb=pre_execute_cb, v3_data=v3_data)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\comfy\custom_nodes\ComfyUI-Lora-Manager\py\metadata_collector\metadata_hook.py", line 177, in async_map_node_over_list_with_metadata
    results = await original_map_node_over_list(
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ...<2 lines>...
    )
    ^
  File "D:\comfy\execution.py", line 318, in _async_map_node_over_list
    await process_inputs(input_dict, i)
  File "D:\comfy\execution.py", line 306, in process_inputs
    result = f(**inputs)
  File "D:\comfy\custom_nodes\ComfyUI_SaveImageWithMetaDataUniversal\saveimage_unimeta\nodes\save_image.py", line 480, in save_images
    pnginfo_dict_src = self.gen_pnginfo(sampler_selection_method, sampler_selection_node_id, civitai_sampler)
  File "D:\comfy\custom_nodes\ComfyUI_SaveImageWithMetaDataUniversal\saveimage_unimeta\nodes\save_image.py", line 883, in gen_pnginfo
    trace_tree_from_this_node = Trace.trace(hook.current_save_image_node_id, hook.current_prompt)
  File "D:\comfy\custom_nodes\ComfyUI_SaveImageWithMetaDataUniversal\saveimage_unimeta\trace.py", line 87, in trace
    if nid not in visited and nid in prompt:  # Ensure the node is not visited and exists
       ^^^^^^^^^^^^^^^^^^
TypeError: unhashable type: 'dict'

[INFO] Prompt executed in 42.05 seconds
```

--- a/saveimage_unimeta/trace.py
+++ b/saveimage_unimeta/trace.py
@@ -17,6 +17,7 @@ from .defs import CAPTURE_FIELD_LIST
 from .defs.combo import SAMPLER_SELECTION_METHOD
 from .defs.meta import MetaField
 from .defs.samplers import SAMPLERS
+from .defs.validators import _is_link_input
 
 from .utils.color import cstr
 
@@ -82,13 +83,8 @@ class Trace:
             current_node_id, distance = node_queue.popleft()
             input_fields = prompt[current_node_id]["inputs"]
             for value in input_fields.values():
-                if isinstance(value, list):
+                if _is_link_input(value):
                     nid = value[0]
-                    if not isinstance(nid, str):
-                        # V3/subgraph links reference nodes as dicts (or other
-                        # non-string ids) and are not traceable in the flat prompt
-                        # graph, so skip them rather than crashing on hashing.
-                        continue
                     if nid not in visited and nid in prompt:  # Ensure the node is not visited and exists
                         class_type = prompt[nid]["class_type"]
                         trace_tree[nid] = TraceEntry(distance + 1, class_type)

--- a/tests/test_defs_validators.py
+++ b/tests/test_defs_validators.py
@@ -49,14 +49,14 @@ def test_is_positive_prompt_detects_known_text_encoder():
     assert not validators_mod.is_positive_prompt("1", None, prompt, None, None, None)
 
 
-# A sampler whose conditioning input references a V3 subgraph (dict link) must be
-# skipped safely instead of raising TypeError: unhashable type: 'dict'.
-def test_positive_prompt_skips_subgraph_dict_links():
+# A list-of-dicts widget value (e.g. a LoRA stack) is not a link and must not
+# raise TypeError: unhashable type: 'dict'.
+def test_positive_prompt_skips_list_of_dicts_inputs():
     prompt = {
         "1": {
             "class_type": "KSampler",
             "inputs": {
-                "positive": [{"subgraph": "abc-123", "output": 0}, 0],
+                "positive": [{"name": "foo.safetensors", "strength": 0.5}],
             },
         },
     }

--- a/tests/test_trace_extended.py
+++ b/tests/test_trace_extended.py
@@ -184,16 +184,16 @@ class TestTraceBFS:
         assert "a" in result
         assert "b" in result
 
-    def test_trace_skips_subgraph_dict_links(self, trace_module):
-        """V3 subgraph links (first element is a dict) must not crash BFS traversal."""
-        # Under the ComfyUI V3 schema, an input connected through a subgraph is
-        # serialised with a dict node reference instead of a string id.
+    def test_trace_skips_list_of_dicts_inputs(self, trace_module):
+        """List-of-dicts widget values (e.g. LoRA stacks) must not crash BFS traversal."""
+        # A LoRA stack widget value is a list of dicts, not a [node_id, output_index]
+        # link. Trace.trace must skip it rather than hashing the dict.
         prompt = {
             "save": {
                 "class_type": "SaveNode",
                 "inputs": {
                     "img": ["sampler", 0],
-                    "subgraph_out": [{"subgraph": "abc-123", "output": 0}, 0],
+                    "lora_stack": [{"name": "foo.safetensors", "strength": 0.5}],
                 },
             },
             "sampler": {"class_type": "KSampler", "inputs": {}},
@@ -201,7 +201,7 @@ class TestTraceBFS:
 
         result = trace_module.Trace.trace("save", prompt)
 
-        # The dict link is not traceable in the flat prompt graph and is skipped.
+        # The list-of-dicts value is not a link and is skipped.
         assert result["save"].distance == 0
         assert result["sampler"].distance == 1
         assert len(result) == 2

